tests/test_mig_shared_install.py

```diff
@@ -237,23 +237,24 @@ def test_creates_output_files_with_permanent_freeze(self):
         expected_generated_dir = cleanpath('confs-stdlocal', self)
         symlink_path = temppath('confs', self)
 
-        generate_confs(
-            self.output_path,
-            destination=symlink_path,
-            destination_suffix='-stdlocal',
-            permanent_freeze=('foo', 'bar', 'baz'),
-            _getpwnam=create_dummy_gpwnam(4321, 1234),
-        )
+        for arg_val in ('yes', 'no', 'foo bar baz'):
```

Comment:
Well, I know how you feel about dynamic code in unit tests and at least I agree that we should keep it simple when coding there - try hard to limit it to cases where looping on *constant* input values prevents significant code duplication. It's a always a trade-off of course.
I'm not so worried about missing secondary failures at first as long as the test always shouts about failing on any regression. So that we will quickly be told about additional regressions when we address one. Yet, I wouldn't complain if someone would split this test case into one for simple boolean values and one for space-separated strings :-)

Reply:
Hmm ok - couple of things. First is since this didn't really do anything in this PR it actually made the diff somewhat noisy to read and ultimately this would have been better introduced at the point we actually needed it :) for the record I had been trying to make user_dict opaque, and this starts to expose it in a way I had been trying to avoid which is why I'd pushed back. Again, I might be wrong about being able to avoid accessing it, but this may end up with another commit that undoes this later which feels a bit like churn. Anyway, didn't really change much in the end.